pkg/rtc/participant.go

```diff
@@ -1264,6 +1261,17 @@ func (p *ParticipantImpl) SetMigrateState(s types.MigrateState) {
 			// re-resolve to check if the track is still active and unsubscribe if none
 			// is active, as local track is in the process of completing publish,
 			// the check would have resolved to an empty track leading to unsubscription.
+			go func() {
+				for {
+					if !p.hasPendingMigratedTrack() {
```

Comment:
would this eventually return false if the publish/migration times out? just to make sure we don't leave anything hanging.

maybe we should check for the connection state here too?

Reply:
Yeah, will add that.

Reply:
Adding a time out too.